samples/bluetooth/mesh/sensor_server/src/model_handler.c

```diff
@@ -737,46 +922,40 @@ static void button_handler_cb(uint32_t pressed, uint32_t changed)
 					     &present_amb_light_level, &val);
 		if (err) {
 			printk("Error publishing present ambient light level (%d)\n", err);
-		} else {
-			printk("Publishing present ambient light level %s\n",
-			       bt_mesh_sensor_ch_str(&val));
 		}
 	}
-
 	if (BUTTON_PRESSED(pressed, changed, BIT(1))) {
-		int64_t thres_micros;
-		enum bt_mesh_sensor_value_status status;
-
-		status = bt_mesh_sensor_value_to_micro(&pres_mot_thres, &thres_micros);
-		if (bt_mesh_sensor_value_status_is_numeric(status)) {
-			k_work_reschedule(&presence_detected_work, K_MSEC(thres_micros / 10000));
-			printk("Presence will be published in %d ms\n",
-			       (uint16_t)(thres_micros / 10000));
-		} else {
-			/* Value is not known, register presence immediately */
-			k_work_reschedule(&presence_detected_work, K_NO_WAIT);
-		}
+		presence_button_handler(true);
 	}
-
 	if (BUTTON_RELEASED(pressed, changed, BIT(1))) {
-		if (!pres_detect) {
-			printk("Publishing presence is aborted\n");
-			k_work_cancel_delayable(&presence_detected_work);
-		} else {
-			int err;
-			struct bt_mesh_sensor_value val = BOOLEAN_INIT(false);
+		presence_button_handler(false);
+	}
+	if (BUTTON_PRESSED(pressed, changed, BIT(2))) {
+		motion_button_handler(true);
+	}
+	if (BUTTON_RELEASED(pressed, changed, BIT(2))) {
+		motion_button_handler(false);
+	}
+	if (BUTTON_PRESSED(pressed, changed, BIT(3))) {
+		int err;
+		static int people_count_idx;
+		struct bt_mesh_sensor_value val;
 
-			err = bt_mesh_sensor_srv_pub(&occupancy_sensor_srv, NULL,
-						&presence_sensor, &val);
+		dummy_people_count_value = dummy_people_count_values[people_count_idx++];
+		people_count_idx = people_count_idx % ARRAY_SIZE(dummy_people_count_values);
 
-			if (err) {
-				printk("Error publishing end of presence (%d)\n", err);
-			} else {
-				printk("Publishing end of presence\n");
-			}
+		err = bt_mesh_sensor_value_from_micro(people_count_sensor.type->channels[0].format,
+						      BASE_UNITS_TO_MICRO(dummy_people_count_value),
+						      &val);
+		if (err) {
```

Comment:
1. I think this should check `if (err && err != -ERANGE)` here. `-ERANGE` is returned when the value is clamped, and I see no reason why clamping should be a problem in this case.
2. If an error other than `-ERANGE` happens, `val` will still be uninitialized, and shouldn't be published

Reply:
These go for every case of converting `bt_mesh_sensor_value_from_...` earlier in the file as well. They should probably all accept -ERANGE, and should return if any other error occurs.